### PR TITLE
fix bn layout network config caching issue

### PR DESCRIPTION
**Pull request by bghimireamd**
pytorch's batch norm was failing because incorrect layout (NCHW was being cached when NHWC). This PR is to fix the issue.

### CODE REVIEW

**Comment by aserio** [comment]
@junliume @DrizztDoUrden, can we get this PR reviewed quickly? It is blocking a feature that needs to be merged ASAP.

File: src/include/miopen/batchnorm/problem_description.hpp [hunk 1]
```diff
@@ -222,10 +222,47 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase, Prob
     std::string in_layout  = "NCHW";
     std::string out_layout = "NCHW";
     std::string din_layout = "NCHW";
+    int spatial_dim        = 2;
```
Comment (averinevg) [inline src/include/miopen/batchnorm/problem_description.hpp]: Looks like the number of dimensions is now hardcoded. How will  3d cases be handled?

File: src/include/miopen/batchnorm/problem_description.hpp [hunk 2]
```diff
@@ -222,10 +222,47 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase, Prob
...
+    {
+        if(spatial_dim == 2)
+        {
+            return dxDesc.GetLayout("NCHW");
+        }
+        else
+        {
+            return dxDesc.GetLayout("NCDHW");
+        }
+    }
```
Comment (averinevg) [inline src/include/miopen/batchnorm/problem_description.hpp]:
```suggestion
    std::string ComputeLayout(const TensorDescriptor& td) const
    {
        if(spatial_dim == 2)
        {
            return td.GetLayout("NCHW");
        }
        else
        {
            return td.GetLayout("NCDHW");
        }
    }

    std::string ComputeInLayout() const
    {
        return ComputeLayout(xDesc);
    }

    std::string ComputeOutLayout() const
    {
        return ComputeLayout(yOrDyDesc);
    }

    std::string ComputeDinLayout() const
    {
        return ComputeLayout(dxDesc);
    }
```

**Review by averinevg — CHANGES_REQUESTED** [review]

**Review by averinevg — APPROVED** [review]
LGTM

**Review by junliume — APPROVED** [review]
